packages/docs/celo-codebase/protocol/identity/phone-number-privacy.md

```diff
@@ -0,0 +1,38 @@
+
+## Decentralized PGPNP
+### Distributed Key Generation
+For the sake of user privacy, no single party should have the ability to unilaterally compute the hash for a given number. With this in mind, we've designed the PGPNP service to be decentralized across a handful of reputable participants. Before the PGPNP service is deployed, a set of operators participated in a Distributed Key Generation (DKG) ceremony to generate a shared secret. You can find the repository for the [DKG setup here](https://github.com/celo-org/celo-threshold-bls-rs). Each instance of the PGPNP service holds a share of the key which can be used to sign the response to the user. When enough of these signatures are combined, their combination can be used to derive the phone number salt. The number of key holders and threshold of signatures required are both configurable at the time of the DKG ceremony.
+
+### Blinding
+When the user queries the phone number, the mobile application first blinds the phone number locally. After the application receives the response, it unblinds it to retrieve the salt. This blinding process preserves the privacy of the mobile number such that the PGPNP cannot tell which number it's providing a salt for; reducing risk of targeted censorship and further increasing privacy.  
+
+### Combiner
+Due to the multi-service communication that needs to happen as part of the K of M signing, we've created a combiner service which performs the orchestration on the user's behalf. This is entirely optional to facilitate the orchestration and a given client can choose to operate their own combiner if desired. Because the combiner only receives the blinded phone number, the combiner cannot tell which number it's providing orchestrating services for. The combiner additionally validates the response from each signer to ensure a corrupt signer cannot corrupt the resulting salt.
+
+## Authentication
+In order to measure the quota for a given request, the PGPNP must check the querier's account information on the Celo network. To prove ownership over the account, the header of the API request contains the signed message body. When the PGPNP receives the request, it authenticates the user by recovering the message signer from the header and comparing it to the value in the message body.
+
+## Request Flow Diagram
+![request flow diagram](https://storage.googleapis.com/celo-website/docs/pgpnp-flow.svg)
+
+## Architecture
+![architecture diagram](https://storage.googleapis.com/celo-website/docs/pgpnp-architecture.jpg)
+
+The hosted architecture is divided into two components, the Combiner and the Signer. Both are hosted as serverless cloud functions with public API endpoints. Both services rely upon a [BLS signing library](https://github.com/celo-org/blind-threshold-bls-wasm#e1e2f8a) which is compiled to a Web Assembly module. The Signer also leverages a hosted PostgreSQL database for tracking quotas and an Azure Key Vault instance for managing the BLS key securely.
```

Comment:
The cloud functions part will likely change soon. Totally fine if you'd like to include it for now and update later, up to you :)

Reply:
Yeah, I was debating this too. I'll leave it reflect how it is today and update it once we change the architecture.